### PR TITLE
Specify a host and protocol manually for target URL generation 

### PR DESCRIPTION
**Pull request by mbarnett**
Necessary so that DOI background jobs can run properly under Resque.

### CODE REVIEW

File: app/services/doi_service.rb [hunk 1]
```diff
@@ -80,13 +80,17 @@ def self.remove(doi)
   private
 
   def ezid_metadata
+    # deal with the fact that there's no request object when running under resque
```
Comment (murny) [inline app/services/doi_service.rb]: should say sidekiq not resque :trollface:

File: app/services/doi_service.rb [hunk 2]
```diff
@@ -80,13 +80,17 @@ def self.remove(doi)
...
+    url_opts = { id: @item.id,
```
Comment (murny) [inline app/services/doi_service.rb]: Alternatively you can set this jupiter wide from config files (production.rb etc.). Something like:
```
# needed when using Rails.application.routes.url_helpers from a model or worker (see http://stackoverflow.com/a/5456103/545137)
Rails.application.routes.default_url_options = { host: 'era.library.ualberta.ca', port: 443, protocol: 'https' }```

Then you can use url_helpers in models/workers/etc

Comment (mbarnett) [reply app/services/doi_service.rb]: Yeah, I guess there's no reason not to do that, as long as it's loaded by Sidekiq when the job runs

Comment (murny) [reply app/services/doi_service.rb]: Should be loaded by sidekiq as they run the same code, done it before on a past project by just plopping it in production.rb etc. 

Initializer also works for example this is what gitlabhq does: https://github.com/gitlabhq/gitlabhq/blob/master/config/initializers/default_url_options.rb

File: config/environments/production.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+Rails.application.routes.default_url_options = { host: 'era.library.ualberta.ca', port: 443, protocol: 'https' }
```
Comment (murny) [inline config/environments/production.rb]: Should add to the other environments too?

test/dev can be like:
Rails.application.routes.default_url_options = { host: 'localhost', port: 3000 }

Comment (mbarnett) [reply config/environments/production.rb]: Specifying a port actually breaks system tests, but I can specify the host in the other environments at least.

**Pull request merged**